jsonb-generator/src/main/java/io/avaje/jsonb/generator/Processor.java

```diff
@@ -60,13 +66,38 @@ public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment
     writeEnumAdapters(round.getElementsAnnotatedWith(element(ValuePrism.PRISM_TYPE)));
     writeAdaptersForMixInTypes(round.getElementsAnnotatedWith(element(JSON_MIXIN)));
     writeAdaptersForImported(round.getElementsAnnotatedWith(element(JSON_IMPORT)));
+    registerCustomAdapters(round.getElementsAnnotatedWith(element(CustomAdapterPrism.PRISM_TYPE)));
     initialiseComponent();
     cascadeTypes();
     writeComponent(round.processingOver());
     return false;
   }
 
-  private void writeEnumAdapters(Set<? extends Element> elements) {
+  private void registerCustomAdapters(Set<? extends Element> elements) {
+
+    for (final var typeElement : ElementFilter.typesIn(elements)) {
+      final var type = typeElement.getQualifiedName().toString();
+      if (CustomAdapterPrism.getInstanceOn(typeElement).isGeneric()) {
+
+        ElementFilter.fieldsIn(typeElement.getEnclosedElements()).stream()
+            .filter(
+                v ->
+                    v.getModifiers().contains(Modifier.STATIC)
+                        && "FACTORY".equals(v.getSimpleName().toString()))
```

Comment:
Hmm, breaking change?  equalsIgnoreCase?

Reply:
existing adapters will be unaffected as this is only for custom adapters annotated with the new annotation